### PR TITLE
Milestone 06 : Views

### PR DESCRIPTION
**Pull request by FaranoshAmini**
## In this pull request, I have implemented the following changes:

- [x] Add the Views associated with the Models and Controllers already created for the Blog App.
- [x] Implement the designs from the sneak peek wireframes.
- [x] Use methods created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<div class="container">
+<% @users.each do |user| %>
+<a href="/users/<%=user.id %>">
+<div class="image">Images</div>
+<div class="card">
+<h1 class="header"><%= user.name%></h1>
+<p class="footer">Number of posts: <%= user.posts_counter %></p>
```
Comment (Owusu-Desmond) [inline app/views/users/index.html.erb]: - [x] Good that you are displaying the user name and user post counter :+1: But please make sure you display the user photo as well using `" alt="<%= user.name%>">` :pray:

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<div class="container">
+<div>
+<div class="image">Images</div>
+<div class="card">
+<h1 class="header"><%=@user.name%></h1>
+<p class="footer">Number of posts: <%=@user.posts_counter %></p>
```
Comment (Owusu-Desmond) [inline app/views/users/show.html.erb]: - [x] Good that you are displaying the user name and user post counter :+1: But please make sure you display the user photo as well using `" alt="<%= user.name%>">` :pray:

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+</div>
+</div>
+<div class="bio">
+<p>Bio:</p>
+<p><%=@user.bio%></p>
+</div>
+<% @user.recent_posts.each do |post| %>
+<div class="post">
+<p><%=post.title %></p>
```
Comment (Owusu-Desmond) [inline app/views/users/show.html.erb]: - [x] Why dont you link the post title to the post-show page like this :point_right: `<h4><%= link_to post.title, user_post_path(post) %></h4>` :blush:

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<div class="container">
+<div class="image">Images</div>
+<div class="card">
+<h1 class="header"><%=@user.name %></h1>
+<p class="footer">Number of posts: <%=@user.posts_counter %></p>
```
Comment (Owusu-Desmond) [inline app/views/posts/index.html.erb]: - [x] Good that you are displaying the user name and user post counter :+1: But please make sure you display the user photo as well using `" alt="<%= user.name%>">` :pray:

File: README.md [hunk 5]
```diff
@@ -55,6 +55,16 @@
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+> Add a link to your deployed project.
+
```
Comment (Owusu-Desmond) [inline README.md]: - [x] Please make sure that you remove this template comment from your readme to make it more professional. :pray:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @FaranoshAmini , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<div class="container">
+<% @users.each do |user| %>
+<a href="/users/<%=user.id %>">
+<div class="image"><img src="<%= @user.photo%>" alt="<%= @user.name%>"></div>
```
Comment (danmainah) [inline app/views/users/index.html.erb]: - [x] @FaranoshAmini great work implementing the changes requested, however, you used @user.photo and @user.name inside a loop resulting in the error below, kindly remove the ` @ ` to remove the error
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @FaranoshAmini , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by FaranoshAmini** [comment]
@danmainah, @ichala, and @Owusu-Desmond Thanks for your kind review and feedback.

**Pull request merged**